test/Constraints/overload.swift

```diff
@@ -287,7 +287,15 @@ func rdar97396399() {
 }
 
 // https://github.com/apple/swift/issues/63834
-func f63834(int: Int, string: String) {} // expected-note{{found candidate with type '(Int, String) -> ()'}}
-func f63834(int: Int, string: Bool) {} // expected-note{{found candidate with type '(Int, Bool) -> ()'}}
-
-let a = f63834(int:string:) as (Int, Int) -> Void // expected-error{{no exact matches in reference to global function 'f63834'}}
+func f63834(int: Int, string: String) {} // expected-note 3{{found candidate with type '(Int, String) -> ()'}}
+// expected-note@-1{{candidate '(Int) -> Void' requires 1 argument, but 2 were provided}}
+// expected-note@-2{{found this candidate}}
+func f63834(int: Int, string: Bool) {} // expected-note 3{{found candidate with type '(Int, Bool) -> ()'}}
+// expected-note@-1{{candidate '(Int) -> Void' requires 1 argument, but 2 were provided}}
+// expected-note@-2{{found this candidate}}
+
+let _ = f63834(int:string:) as (Int, Int) -> Void // expected-error{{no exact matches in reference to global function 'f63834'}}
+let _ = f63834(int:string:) as (Int) -> Void // expected-error{{no exact matches in reference to global function 'f63834'}}
+let _ = f63834(int:string:) as Int // expected-error{{no exact matches in reference to global function 'f63834'}}
+let _ = (f63834(int:string:)) as Int // expected-error{{no exact matches in reference to global function 'f63834'}}
+let _ = f63834(int:string:) as (Double, Double) -> Void // expected-error{{no exact matches in reference to global function 'f63834'}}
```

Comment:
Could you decouple these test cases by the kind of note they produce to make them more robust and clarify which ones produce which notes?

Reply:
Done :)